Gem/Code/Source/RHI/XRExampleComponent.cpp

```diff
@@ -0,0 +1,377 @@
+    const AZ::Vector4 DarkGreen = AZ::Vector4(0, 0.25f, 0, 0);
+    const AZ::Vector4 Blue = AZ::Vector4(0, 0, 1, 0);
+    const AZ::Vector4 DarkBlue = AZ::Vector4(0, 0, 0.25f, 0);
+
+    void XRExampleComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<XRExampleComponent, AZ::Component>()->Version(0);
+        }
+    }
+
+    XRExampleComponent::XRExampleComponent()
+    {
+        m_supportRHISamplePipeline = true;
+    }
+
+    void XRExampleComponent::Activate()
+    {
+        using namespace AZ;
+
+        CreateCubeInputAssemblyBuffer();
+        CreateCubePipeline();
+        CreateScope();
+        AZ::RHI::RHISystemNotificationBus::Handler::BusConnect();
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void XRExampleComponent::OnTick(float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        m_time += deltaTime;
+    }
+
+    void XRExampleComponent::OnFramePrepare(AZ::RHI::FrameGraphBuilder& frameGraphBuilder)
+    {
+        AZ::Matrix4x4 projection = AZ::Matrix4x4::CreateIdentity();
+        AZ::Matrix4x4 vpMat = AZ::Matrix4x4::CreateIdentity();
+
+        AZ::RPI::XRRenderingInterface* xrSystem = AZ::RPI::RPISystemInterface::Get()->GetXRSystem();
+        if (xrSystem && xrSystem->ShouldRender())
+        {
+            AZ::RPI::FovData fovData = xrSystem->GetViewFov(m_viewIndex);
+            AZ::RPI::PoseData poseData = xrSystem->GetViewPose(m_viewIndex);
+                
+            static const float clip_near = 0.05f;
+            static const float clip_far = 100.0f;
+            projection = xrSystem->CreateProjectionOffset(fovData.m_angleLeft, fovData.m_angleRight, 
+                                                          fovData.m_angleDown, fovData.m_angleUp, 
+                                                          clip_near, clip_far);
+
+            AZ::Matrix4x4 viewMat = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(poseData.orientation, poseData.position);;
+            viewMat.InvertTransform();
+            m_viewProjMatrix = projection * viewMat;
+ 
+            AZ::Matrix4x4 initialScaleMat = AZ::Matrix4x4::CreateScale(AZ::Vector3(0.1f, 0.1f, 0.1f));
+
+            //Model matrix for the cube related to the front view
+            AZ::RPI::PoseData frontViewPoseData = xrSystem->GetViewFrontPose();
+            m_modelMatrices[0] = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(frontViewPoseData.orientation, frontViewPoseData.position) * initialScaleMat;
+                      
+            //Model matrix for the cube related to the left controller
+            AZ::RPI::PoseData controllerLeftPose = xrSystem->GetControllerPose(0);
+            AZ::Matrix4x4 leftScaleMat = initialScaleMat * AZ::Matrix4x4::CreateScale(AZ::Vector3(xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0)));
```

Comment:
`AZ::Vector3(float)` constructor sets the same value to the 3 components, so:

`AZ::Vector3(xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0), xrSystem->GetControllerScale(0))`

can be just

`AZ::Vector3(xrSystem->GetControllerScale(0))`